src/Controller/UserController.php

```diff
@@ -64,4 +78,64 @@ public function register(UserDto $userDto): Response
 
         return new JsonResponse($savedDto, Response::HTTP_CREATED);
     }
+
+    /**
+     * @Route(path="/delete/{email}", methods={"DELETE"})
+     */
+    public function softDelete(string $email): Response
+    {
+        $user = $this->userRepository->findOneBy(['email' => $email]);
+
+        if (null === $user) {
+            $this->logger->info('Soft delete failed: no account for such email');
+
+            return new JsonResponse('No account associated with email', Response::HTTP_NOT_FOUND, [], true);
+        }
+        $this->eventDispatcher->dispatch(new UserSoftDeleteEvent($user), UserSoftDeleteEvent::NAME);
+
+        $this->logger->info('Account soft deleted: ' . $email);
+
+        return new JsonResponse('Account deleted.', Response::HTTP_FOUND, [], true);
+    }
+
+    /**
+     * @Route(path="/recover/{email}", methods={"GET"})
+     */
+    public function softRecover(string $email): Response
+    {
+        $this->entityManager->getFilters()->disable('softdeleteable');
+        $recoveredUser = $this->userRepository->findOneBy(['email' => $email]);
```

Comment:
if you want to have the best naming here that indicates the state of the user, please go for `userToBeRecovered`